ldms/src/sampler/spank/slurm_notifier.c

```diff
@@ -800,25 +892,31 @@ slurm_spank_task_exit(spank_t sh, int argc, char *argv[])
 
 int slurm_spank_exit(spank_t sh, int argc, char *argv[])
 {
-	/* Runs as root */
 	spank_context_t context = spank_context();
-	const char *context_str;
-	jbuf_t jb;
-
-	if (0 == nnodes(sh))
-		/* Ignore events before node assignment */
-		return ESPANK_SUCCESS;
+	jbuf_t jb = NULL;
 
 	switch (context) {
+	case S_CTX_ALLOCATOR:
```

Comment:
Again, S_CTX_ALLOCATOR should not be used at all, and this case statement can be replaced with a conditional using S_CTX_REMOTE.